src/annotator/integrations/vitalsource.ts

```diff
@@ -436,7 +505,23 @@ export class VitalSourceContentIntegration
       return selectors;
     }
 
-    const pageInfo = await this._bookElement.getCurrentPage();
+    const [pageInfo, toc] = await Promise.all([
+      this._bookElement.getCurrentPage(),
+      this._bookElement.getTOC(),
+    ]);
+
+    let title = pageInfo.chapterTitle;
+
+    // Find the first table of contents entry that corresponds to the current page,
+    // and use its title instead of `pageInfo.chapterTitle`. This works around
+    // https://github.com/hypothesis/client/issues/4986.
+    const pageCFI = documentCFI(pageInfo.cfi);
+    const tocEntry = toc.data?.find(
+      entry => documentCFI(entry.cfi) === pageCFI
+    );
+    if (tocEntry) {
+      title = tocEntry.title;
+    }
```

Comment:
As far as I know, there _should_ always be a TOC entry for the current page. It is technically possible that the TOC fails to load (the VS API call to fetch the toc data can be blocked in devtools to test this), and maybe there are cases where the TOC is incomplete. Hence the fallback.